Week2/HYF-JS3-HW2/index.js

```diff
@@ -0,0 +1,124 @@
+    const elem = document.createElement(name);
+    parent.appendChild(elem);
+    Object.keys(options).forEach(key => {
+      const value = options[key];
+      if (key === 'text') {
+        elem.textContent = value;
+      } else {
+        elem.setAttribute(key, value);
+      }
+    });
+    return elem;
+  }
+
+  const root = document.getElementById('root');
+  const header = createAndAppend('header', root, { class: 'header' });
+  createAndAppend('p', header, { text: 'HYF Repositories' });
+
+  const select = createAndAppend('select', header, { id: 'selector' });
+  const option = x => {
+    for (let i = 0; i < x.length; i++) {
+      createAndAppend('option', select, { text: x[i].name, value: i });
+    }
+  };
+
+  const mainContainer = createAndAppend('div', root, { id: 'container' });
+  const leftContainer = createAndAppend('div', mainContainer, { class: 'left-block frame' });
+  const table = createAndAppend('table', leftContainer);
+  const listContributor = createAndAppend('tbody', table);
+
+  const rightContainer = createAndAppend('div', mainContainer, { class: 'right-block frame' });
+  createAndAppend('p', rightContainer, {
+    text: 'Contributions',
+    class: 'contributor',
+  });
+
+  const list = createAndAppend('ul', rightContainer, { class: 'list1' });
+
+  function renderContributors(selRep) {
+    list.innerHTML = '';
+
+    for (let i = 0; i < selRep.length; i++) {
+      const li = createAndAppend('li', list, { class: 'contributor-item' });
+      const linkFor = createAndAppend('a', li, { target: '_blank', href: selRep[i].html_url });
+      createAndAppend('img', linkFor, { src: selRep[i].avatar_url, class: 'avatar', height: 48 });
+      const divCont = createAndAppend('div', linkFor, { class: 'contributor-data' });
+      createAndAppend('div', divCont, { text: selRep[i].login });
+      createAndAppend('div', divCont, {
+        text: selRep[i].contributions,
+        class: 'badge',
+      });
+    }
+  }
+
+  function renderRep(listContributors, repo) {
+    const content = [
+      { title: 'Repository', attribute: 'name' },
+      { title: 'Description', attribute: 'description' },
+      { title: 'Forks', attribute: 'forks' },
+      { title: 'Updated', attribute: 'updated_at' },
+    ];
+    listContributors.innerHTML = '';
+
+    for (let i = 0; i < content.length; i++) {
+      const headTitle = createAndAppend('tr', listContributors);
+      createAndAppend('td', headTitle, { text: `${content[i].title} :`, class: 'label' });
+      const cellContent = createAndAppend('td', headTitle);
+      if (content[i].attribute === 'name') {
+        createAndAppend('a', cellContent, {
+          href: repo.html_url,
+          text: repo.name,
+          target: '_blank',
+        });
+      } else {
+        cellContent.textContent = repo[content[i].attribute];
+      }
+    }
+    fetchJSON(repo.contributors_url).then(contributors => renderContributors(contributors));
```

Comment:
You must also catch the errors here.

Reply:
Main problem is I did not understand the JS3 Week 2 and 3. And more than two months pass on it. First I should remember what I did in the JS3 HW1 and it was not easy. As you mention my coding is not easy to understand what is going on. I study promises today by myself after that try to do homework. :) . I am planing to find a way to study promises deeply.

Reply:
Hi @remarcmij I did not understand this part  

"You also removed some code from the starter file for reasons unclear to me, e.g. function main(), window.onload."

Please can you put more light on it.
Thanks